libs/bits/bit_array.go

```diff
@@ -126,15 +126,14 @@ func (bA *BitArray) Or(o *BitArray) *BitArray {
 	if o == nil {
 		return bA.Copy()
 	}
+	o = o.Copy()
```

Comment:
Maybe this should be AFTER bA.mtx.RLock, to ensure that neither bA nor o changed during execution of this function

Reply:
if to add a condition to check `o` and `bA` on equality, anyway go-deadlock reports about potential deadlock issue